blocks/form/form.js

```diff
@@ -0,0 +1,217 @@
+  const input = document.createElement('textarea');
+  input.id = fd.Field;
+  input.setAttribute('placeholder', fd.Placeholder);
+  if (fd.Mandatory === 'x') {
+    input.setAttribute('required', 'required');
+  }
+  return input;
+}
+
+function createLabel(fd) {
+  const label = document.createElement('label');
+  label.setAttribute('for', fd.Field);
+  label.textContent = fd.Label;
+  if (fd.Mandatory === 'x') {
+    label.classList.add('required');
+  }
+  return label;
+}
+
+function applyRules(form, rules) {
+  const payload = constructPayload(form);
+  rules.forEach((field) => {
+    const {
+      type,
+      condition: { key, operator, value },
+    } = field.rule;
+    if (type === 'visible') {
+      if (operator === 'eq') {
+        if (payload[key] === value) {
+          form.querySelector(`.${field.fieldId}`).classList.remove('hidden');
+        } else {
+          form.querySelector(`.${field.fieldId}`).classList.add('hidden');
+        }
+      }
+    }
+  });
+}
+
+function fill(form) {
+  const { action } = form.dataset;
+  if (action === '/tools/bot/register-form') {
+    const loc = new URL(window.location.href);
+    form.querySelector('#owner').value = loc.searchParams.get('owner') || '';
+    form.querySelector('#installationId').value = loc.searchParams.get('id') || '';
+  }
+}
+
+async function createForm(formURL) {
+  const { pathname } = new URL(formURL);
+  const resp = await fetch(pathname);
+  const json = await resp.json();
+  const form = document.createElement('form');
+  const rules = [];
+  // eslint-disable-next-line prefer-destructuring
+  form.dataset.action = pathname.split('.json')[0];
+  json.data.forEach((fd) => {
+    fd.Type = fd.Type || 'text';
+    const fieldWrapper = document.createElement('div');
+    const style = fd.Style ? ` form-${fd.Style}` : '';
+    const fieldId = `form-${fd.Type}-wrapper${style}`;
+    fieldWrapper.className = fieldId;
+    fieldWrapper.classList.add('field-wrapper');
+    switch (fd.Type) {
+      case 'select':
+        fieldWrapper.append(createLabel(fd));
+        fieldWrapper.append(createSelect(fd));
+        break;
+      case 'heading':
+        fieldWrapper.append(createHeading(fd));
+        break;
+      case 'checkbox':
+        fieldWrapper.append(createInput(fd));
+        fieldWrapper.append(createLabel(fd));
+        break;
+      case 'text-area':
+        fieldWrapper.append(createLabel(fd));
+        fieldWrapper.append(createTextArea(fd));
+        break;
+      case 'submit':
+        fieldWrapper.append(createButton(fd));
+        break;
+      default:
+        // For now, we are not creating any label for the email field
+        //   fieldWrapper.append(createLabel(fd));
+        fieldWrapper.append(createInput(fd));
+    }
+
+    if (fd.Rules) {
+      try {
+        rules.push({ fieldId, rule: JSON.parse(fd.Rules) });
+      } catch (e) {
+        // eslint-disable-next-line no-console
+        console.warn(`Invalid Rule ${fd.Rules}: ${e}`);
```

Comment:
What are the `Rules`?

Reply:
We're not using Rules for this form. Removed it.